Sources/HAP/Base/Generated.swift

```diff
@@ -96,6 +137,61 @@ public extension ServiceType {
 	static let windowCovering = ServiceType(0x008C)
 }
 
+extension ServiceType: CustomStringConvertible {
+    public var description: String {
```

Comment:
I would prefer a switch-statement here instead with cases similar to the definition, like so:
```swift
    public var description: String {
        switch self {
        case let .appleDefined(hex):
            switch hex {
            case 0x0239: return "Accessory Runtime Information"
            case 0x00BB: return "Air Purifier"
            // ...
            default: fatalError("Unknown Apple-defined service type '\(hex)'")
            }
        case let .custom(uuid):
            return "Other \(uuid)"
        }
    }
```

Reply:
OK I have added the top level switch, but a switch and fatalError is not appropriate for the descriptions themselves. It is quite possible that HAP does not have the full knowledge of all valid values.

Reply:
There's only a `fatalError` for the `appleDefined` case. But I agree that you could put a value in there that we're unaware of. So in that case something like `"Apple-defined \(hex)"` would be fine.

Reply:
Sure, that's in my commit of yesterday

Reply:
I still think it should use a switch, like my suggested code. Instead of a fatalError you could return the value in the default case.

Reply:
Ah I think there was some filter active as I didn't see those changes.

Reply:
It's definitely there in commit 64bd0f2cb12418813ddcc81b72edf0f6beb62dba

for both ServiceType and CharacteristicType